pkg/engine/engine.go

```diff
@@ -1016,25 +1016,27 @@ func (cad *cadEngine) UpdatePackageResources(ctx context.Context, repositoryObj
 	resources := repository.PackageResources{
 		Contents: prevResources.Spec.Resources,
 	}
-	appliedResources, _, err := applyResourceMutations(ctx, draft, resources, mutations)
+
+	appliedResources, renderStatus, err := applyResourceMutations(ctx, draft, resources, mutations)
```

Comment:
are you using the `renderStatus` return value somewhere? if not, then I think marking the fact that it is ignored by naming it `_` is the idiomatic way to go.

Reply:
`renderStatus` is returned by the function on lines 1043 and 1047.

Now, `renderStatus` was set by line 1032 before I introduced the if statement at line 1025. Looking at the `applyResourceMutations` function, when it is called at line 1020, it will return with a value of `nil`; the loop exits on the `continue` at line 1057 with `renderStatus` not set.

Another approach would be to use `-` in line 1021 as you sugges, but then I would have to add a line before line 1025 to declare `renderStatus` and set it to nil for the return statements.

I think the code here is quite confusing, especially the double call to `applyResourceMutations` but here, I was trying to make as few changes as possible. I think engine.go in its entirity should be among the first files to target for refactoring in Porch.

Reply:
The `renderStatus` returned by this method will eventually be written back to the status of the `PackageRevisionResources` object returned by the Update() call. So far it represented the results of the render operation (a.k.a. executing the pipeline of the kpt package).
If the body of your if statement is skipped, no actual rendering is happening, so IMHO the intuitive way of handling this case would be to return with a nil renderStatus. 

Now I think this is exactly what your code is doing, because the first `applyResourceMutations` call doesn't have a render mutation in its input (a render mutation is of type "eval"), so it will return with a nil renderStatus. So I think your code is correct.

However.... :) [sorry for the nitpicking]
IMHO it would be more readable if you would explicitly drop the renderStatus return value in the first `applyResourceMutations` call, by naming it `_`, as it was before. 
And explicitly set the renderStatus to nil in the else branch of your new if statement. That would express the logic more directly: if there was a change do a render and return with its results, if there are no changes, no render is needed, and return with and empty render result.

I just want to reiterate that your code is correct as is, so you can safely choose to ignore any of the above.

Reply:
I'll change it, it's probably clearer all right, I'll update it.

Reply:
It's changed in the latest update there. Thanks @kispaljr